python-avd/pyavd/_eos_designs/structured_config/underlay/router_isis.py

```diff
@@ -32,23 +32,23 @@ def router_isis(self: AvdStructuredConfigUnderlay) -> dict | None:
             "instance": self.shared_utils.isis_instance_name,
             "log_adjacency_changes": True,
             "net": self._isis_net,
-            "router_id": self.shared_utils.router_id if not self.shared_utils.use_router_general_for_router_id else None,
+            "router_id": self.shared_utils.router_id if not self.inputs.use_router_general_for_router_id else None,
             "is_type": self._is_type,
-            "address_family_ipv4": {"enabled": True, "maximum_paths": get(self._hostvars, "isis_maximum_paths", default=4)},
+            "address_family_ipv4": {"enabled": True, "maximum_paths": self.inputs.isis_maximum_paths},
         }
 
         if self.shared_utils.underlay_ldp is True:
             router_isis["mpls_ldp_sync_default"] = True
 
         # TI-LFA
-        if get(self._hostvars, "isis_ti_lfa.enabled") is True:
+        if self.inputs.isis_ti_lfa.enabled:
             router_isis["timers"] = {
                 "local_convergence": {
-                    "delay": get(self._hostvars, "isis_ti_lfa.local_convergence_delay", default="10000"),
+                    "delay": self.inputs.isis_ti_lfa.local_convergence_delay,
                     "protected_prefixes": True,
                 },
             }
-        ti_lfa_protection = get(self._hostvars, "isis_ti_lfa.protection")
```

Comment:
```
        match self.inputs.isis_ti_lfa.protection:
            case "link":
                router_isis["address_family_ipv4"]["fast_reroute_ti_lfa"] = {"mode": "link-protection"}
            case "node":
                router_isis["address_family_ipv4"]["fast_reroute_ti_lfa"] = {"mode": "node-protection"}
```

could even do with our valid values

```
        if self.inputs.isis_ti_lfa.protection:
            router_isis["address_family_ipv4"]["fast_reroute_ti_lfa"] = {"mode": f"{self.inputs.isis_ti_lfa.protection}-protection"}
```

Reply:
I don't feel comfortable with that approach right now. Maybe once I get Literal type hints based on valid values :)